pkg/ec/error.go

```diff
@@ -26,11 +26,12 @@ import (
 )
 
 type ControllerError struct {
-	statusCode   int
-	cause        string
-	resourceType string
-	err          error
-	Event        interface{}
+	statusCode          int
+	retryDelayInSeconds int
```

Comment:
Does a negative value make sense here? Maybe unsigned?

Reply:
Excellent idea! `time.Duration` would be best, then I can drop the units too!